File: rb/features.rb
# Usercheck SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module UsercheckFeatures
  def self.make_feature(name)
    case name
    when "base"
      UsercheckBaseFeature.new
    when "ratelimit"
      UsercheckRatelimitFeature.new
    when "retry"
      UsercheckRetryFeature.new
    when "test"
      UsercheckTestFeature.new
    when "timeout"
      UsercheckTimeoutFeature.new
    else
      UsercheckBaseFeature.new
    end
  end
end

File: py/features.py
# Usercheck SDK feature factory

from feature.base_feature import UsercheckBaseFeature
from feature.test_feature import UsercheckTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UsercheckBaseFeature(),
        "test": lambda: UsercheckTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
